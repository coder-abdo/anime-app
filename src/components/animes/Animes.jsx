import React from "react";
import { Anime } from "../anime/Anime";
import { Container } from "../../sharedStyle/Container";
export const Animes = ({ animes }) => {
  return (
    <Container anime="anime" data-testid="animes">
      {animes.map(({ mal_id, image_url, url, score, title }) => (
        <Anime
          key={mal_id}
          image={image_url}
          title={title}
          score={score}
          url={url}
          id={mal_id}
          data-testid="animes"
        />
      ))}
    </Container>
  );
};
