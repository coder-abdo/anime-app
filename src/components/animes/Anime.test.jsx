import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Router } from "react-router";
import { createBrowserHistory } from "history";
import { Animes } from "./Animes";

afterEach(cleanup);
test("render Animes Component by passing suitable props", () => {
  const history = createBrowserHistory();
  const mockANimesList = [
    {
      mal_id: "123-56hg",
      image_url: "image.jpg",
      title: "dummy title",
      score: 4.5,
      url: "https://www.anime-url.com"
    },
    {
      mal_id: "987-vnhg",
      image_url: "image2.jpg",
      title: "dummy2 title",
      score: 4.5,
      url: "https://www.anime-url.com"
    }
  ];
  const { container, asFragment } = render(
    <Router history={history}>
      <Animes animes={mockANimesList} />
    </Router>
  );
  expect(container).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
