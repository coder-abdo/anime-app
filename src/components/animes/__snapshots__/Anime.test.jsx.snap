// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Animes Component by passing suitable props 1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ jomUug"
    data-testid="animes"
  >
    <div
      class="sc-AxjAm gItdEj"
    >
      <div
        class="more-detail"
      >
        <a
          class="more-link"
          data-testid="more-details"
          href="/123-56hg"
        >
          more details
        </a>
      </div>
      <div
        class="imgContainer"
      >
        <img
          alt="dummy title"
          src="image.jpg"
        />
      </div>
      <div
        class="details"
      >
        <h3
          data-testid="anime-title"
        >
          dummy title
        </h3>
        <footer>
          <h4
            data-testid="anime-score"
          >
            score: 4.5
          </h4>
          <a
            href="https://www.anime-url.com"
          >
            myAnime list page
          </a>
        </footer>
      </div>
    </div>
    <div
      class="sc-AxjAm gItdEj"
    >
      <div
        class="more-detail"
      >
        <a
          class="more-link"
          data-testid="more-details"
          href="/987-vnhg"
        >
          more details
        </a>
      </div>
      <div
        class="imgContainer"
      >
        <img
          alt="dummy2 title"
          src="image2.jpg"
        />
      </div>
      <div
        class="details"
      >
        <h3
          data-testid="anime-title"
        >
          dummy2 title
        </h3>
        <footer>
          <h4
            data-testid="anime-score"
          >
            score: 4.5
          </h4>
          <a
            href="https://www.anime-url.com"
          >
            myAnime list page
          </a>
        </footer>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
